Examples/Detectors/MagneticField/include/ActsExamples/MagneticField/ToroidalField.hpp

```diff
@@ -0,0 +1,128 @@
+// This file is part of the ACTS project.
+//
+// Copyright (C) 2016 CERN for the benefit of the ACTS project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Definitions/Algebra.hpp"
+#include "Acts/MagneticField/MagneticFieldContext.hpp"
+#include "Acts/MagneticField/MagneticFieldProvider.hpp"
+#include "Acts/Utilities/Result.hpp"
+
+#include <array>
+#include <vector>
+
+namespace Acts {
+
+class ToroidalField final : public MagneticFieldProvider {
+ public:
+  struct BarrelConfig {
+    float R_in = 4.9f;    // [m]
+    float R_out = 10.0f;  // [m]
+    float c = 25.3f;      // [m]
+    float b = 0.16f;      // [m]
+    float I = 20500.0f;   // [A]
+    int Nturns = 120;     // [turns]
```

Comment:
For the units: does it make sense to use the unit system that ACTS ships with?

E.g., in the code interpreting these numbers you would just do

```cpp
BarrelConfig c = myConfig();

c.c / 1_m; // <- this gives you `c` in meters, regardless of what the user put in
```